docs/notary-server-config.md

```diff
@@ -0,0 +1,160 @@
+		means that the server will not be acessible except locally (via SSH
+		tunnel, or just on a local terminal).
+
+- `tls_key_file` and `tls_cert_file` - **optional**, but must be provided together
+	if provided at all.  These specify the private key and TLS cert,
+	respectively, to use for HTTPS.  If not provided, the server will use HTTP
+	instead.
+
+## `trust service` section (optional)
+
+The trust service section must be included in order to specify a remote trust
+service.  If it is left out, a local in-memory trust service will be used.
+
+Remote trust service example:
+
+```json
+"trust_service": {
+	"type": "remote",
+	"hostname": "notarysigner",
+	"port": "7899",
+	"key_algorithm": "ecdsa",
+	"tls_ca_file": "./fixtures/root-ca.crt",
+	"tls_client_key": "./fixtures/notary-server.key",
+	"tls_client_cert": "./fixtures/notary-server.crt"
+}
+```
+
+- `type` - **required** - must be `remote` or `local`
+- `hostname` - **required** - the hostname of the remote trust service
+- `port` - **required** - the GRPC port of the remote trust service
+- `key_algorithm` - **required** - the key algorithm/type that will be generated
+	and stored on the signing service.  Valid values are `ecdsa`, `rsa`, and
+	`ed25519`.
+- `tls_ca_file` - **optional** - the root CA or TLS cert of the remote service,
+	it is self-signed or otherwise not in the system's default trust roots
+- `tls_client_key`, `tls_client_cert` - **optional**, but must be provided
+	together if provided at all.  These specify the private key and client cert,
+	respectively, to use for TLS mutual authentication.  This is required if
+	the trust services requires mutual authentication.
+
+
+## `logging` section (optional)
+
+The logging section sets the log level of the server.  If not provided, or if
+any part of this section is invalid, the server defaults to an ERROR logging
+level.
+
+Example:
+
+```json
+"logging": {
+	"level": "debug"
+}
+```
+
+- `level` - **required** - one of `"debug"`, `"info"`, `"warning"`, `"error"`,
+	`"fatal"`, or `"panic"`
+
+## `storage` section (optional)
+
+The storage section sets the storage options for the server.  If not provided,
+an in-memory store will be used.
+
+DB storage example:
+
+```json
+"storage": {
+	"backend": "mysql",
+	"db_url": "dockercondemo:dockercondemo@tcp(notarymysql:3306)/dockercondemo"
+}
+```
+
+- `backend` - **required** - currently, only a value of `"mysql"` is supported
+- `db_url` - **required** - the URL used to access the DB, which includes both
+	the endpoint and the username/credentials
+
+## `reporting` section (optional)
+
+The reporting section contains any configuration for reporting errors, etc. to
+services via [logrus hooks](https://github.com/Sirupsen/logrus).  Currently the
+only supported services is [Bugsnag](https://bugsnag.com).  (See
+[bugsnag-go](https://github.com/bugsnag/bugsnag-go/) for more information about
+configuration.
+
+```json
+"reporting": {
+	"bugsnag": "yes",
```

Comment:
Should we make this a subtree instead?  e.g.:

```json
"reporting" : {
    "bugsnag": {
    ...
    }
}
```

Does it need to remain backwards compatible?